### DISCUSSION TITLE
How to configure template to show entity in HA?

### DISCUSSION THREAD

**Discussion by Dexter0007** [arendst/Tasmota#21801] [Q&A]
Hi, I'm trying to configure my custom device by template for display binary_sensor or output in Home Assistant in Tasmota integration (NOT MQTT).
If I set GPIO25 as Relay that Switch appears in HA, but if I set anything else, the entity disappears.
How can I get entity with ON/OFF state in HA ?

![image](https://github.com/user-attachments/assets/13adb66b-2678-45d5-8b32-87dad1766d5b)
![image](https://github.com/user-attachments/assets/bec42de4-a5a5-4b42-8857-067893a339bd)

**Comment by sfromis** [comment]
The way is to config the led as a "Relay" pin, as you already saw working. Or maybe as a PWM, to also have variable brightness. Pins configured as a "Led" are local status leds, not intended to be seen as separate features.

**Comment by Dexter0007** [comment]
The problem is that the pin set as Relay can be switched from HA.  In my device, GPIO25 is supposed to be the alarm signal (read only) from the device.

**Reply by sfromis** [reply]
Well, what you posted was for `Led` which is output also, thus not clear what you wanted.

As I'm not using HA, I can't suggest there, but the Tasmota docs does have some hints about binary sensors.
https://tasmota.github.io/docs/Home-Assistant/#binary-sensors
